### DISCUSSION TITLE
运行时配置layout怎么在组件控制菜单折叠

### DISCUSSION THREAD

**Discussion by Ivan-YangSj** [umijs/umi#12364] [Q&A]
在app.ts中配置layout，如何在业务组件控制菜单折叠

**Comment by fz6m** [comment]
因为 max 自带的 layout 本质上是 [`<ProLayout />`](https://procomponents.ant.design/components/layout) ，你在它的文档里找一下有没有控制这方面的配置吧，之后在运行时 `export const layout` 里就可以配。

**Comment by jiajiaxing1** [comment]
具体需要实现的场景能大概描述下吗
是在页面上点击某个按钮需要展开或者折叠某个菜单吗
如果是这种场景的话 写了个demo可以参考下

`
 export const layout = () => {
	const [openKeys, setOpenKeys] = useState([]);
	return {
		// 菜单配置
		menuProps: {
			openKeys,
			onOpenChange: (openKeys: string[]) => {
				setOpenKeys(openKeys);
			},
		},

		childrenRender: (children) => {
			return (
				<ConfigProvider>
					{/* 这里把控制菜单的openKeys传给子组件
					子组件通过  const { setOpenKeys } = useOutletContext();调用
					*/}
					<Outlet context={{ setOpenKeys }} />
				</ConfigProvider>
			);
		},
	};
};
`
